### PR TITLE
055 - Migrated to node.js test runner

### PR DESCRIPTION
**Pull request by ThotakuraHarish2003**
Can u pls check my code im new to testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "scripts": {
     "build": "tsc",
     "prepublishOnly": "npm run build",
-    "test": "mocha",
+    "test": "node --no-warnings --loader tsx --test test/api.spec.ts",
```
Comment (fraxken) [inline package.json]: Should be enough here:

`node --loader tsx --test test/`

File: test/api.spec.ts [hunk 2]
```diff
@@ -30,22 +28,34 @@ describe("downloads", () => {
   it("should throw error if pkg is not defined", async() => {
     const undefinedPkg = undefined;
 
-    await expect(downloads(undefinedPkg as any))
```
Comment (fraxken) [inline test/api.spec.ts:33]: Node.js assert module have an equivalent `rejects` method

https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message

File: test/registry.spec.ts [hunk 3]
```diff
@@ -21,35 +21,41 @@ describe("NPM Registry", () => {
...
 
     it("should throw if the string URL is invalid", () => {
-      expect(() => setLocalRegistryURL("foobar")).to.throw();
+      try{
+        setLocalRegistryURL("foobar");
+      }
+      catch (error) {
+        console.log(error.message)
+        assert.strictEqual(error.message,"Invalid URL")
+      }
```
Comment (fraxken) [inline test/registry.spec.ts:58]: Node.js assert module has a `throws` method

https://nodejs.org/api/assert.html#assertthrowsfn-error-message

**Review by fraxken — CHANGES_REQUESTED** [review]

**Comment by ThotakuraHarish2003** [comment]
oh thanks so should i change it or u will change it :)

File: package.json (hunk 1 above)
Comment (ThotakuraHarish2003) [reply package.json]: cant seem to work tho it dosent detect tests and display there are 0 tests

**Comment by fraxken** [comment]
@ThotakuraHarish2003 You need to do it :)

**Pull request closed without merging**